lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -44,21 +44,36 @@ def object
         def initialize(serializer, options = {})
           super
           @include_tree = IncludeTree.from_include_args(options[:include])
-          @fieldset = options[:fieldset] || ActiveModel::Serializer::Fieldset.new(options.delete(:fields))
+          @fieldset = options[:fieldset] || ActiveModel::Serializer::Fieldset.new(options[:fields])
```

Comment:
Do we officially support the `fieldset` option? (that is, providing an instance of `Fieldset`, rather than a Hash of Arrays of Symbols) 
It seems it was introduced only to handle the recursive call to the adapter for collections of resources ([here](https://github.com/rails-api/active_model_serializers/pull/1300/files#diff-ae1e38e96a8bf28ae37918a65c448c73L85)).